src/Core/Settings.h

```diff
@@ -421,6 +421,7 @@ class IColumn;
     M(UInt64, min_free_disk_space_for_temporary_data, 0, "The minimum disk space to keep while writing temporary data used in external sorting and aggregation.", 0) \
     \
     M(DefaultDatabaseEngine, default_database_engine, DefaultDatabaseEngine::Atomic, "Default database engine.", 0) \
+    M(DefaultTableEngine, default_table_engine, DefaultTableEngine::None, "Default table engine used when ENGINE is not set in CREATE statement.",0) \
```

Comment:
+1, otherwise it will introduce extra complexity to list all possible table engines.

Reply:
> Do we want settings to be used in parser?

No, we should never use Settings/Context in parsers. It's (almost) always possible to apply all required settings after parsing.

> Do we want to add some parsing in Interpreter?

We should avoid it if possible, but sometimes it's ok. Why do you think that we will need to parse something in Interpreter?

Reply:
I thought String as a default_table_engine can be treated as engine definition with parameters. And this will require parsing.
Just name is similar to current version, but we will get error earlier - when we change the setting or start a server. I propose to leave Enum for now